pynadjust/results.py

```diff
@@ -0,0 +1,1989 @@
+        """
+        Method to consume DynAdjust output files [*.adj, *.apu and *.xyz]
+        :return:
+        """
+        if self.xyz_file:
+            read_xyz_file(
+                xyz_file=self.xyz_file,
+                stns=self.stns,
+                adj_metadata=self.adj_metadata,
+                file_metadata=self.file_metadata
+            )
+
+        if self.adj_file:
+            # clear msrs before (re)loading to avoid doubling up.
+            self.msrs = []
+
+            read_adj_file(
+                adj_file=self.adj_file,
+                stns=self.stns,
+                msrs=self.msrs,
+                adj_metadata=self.adj_metadata,
+                file_metadata=self.file_metadata,
+                adj_stats=self.adj_stats
+            )
+
+            # tally total adj measurements
+            self.count_adj_msrs()
+
+        if self.apu_file:
+            read_apu_file(
+                apu_file=self.apu_file,
+                stns=self.stns,
+                adj_metadata=self.adj_metadata,
+                file_metadata=self.file_metadata,
+                adj_stats=self.adj_stats
+            )
+
+    def link_source_with_msr_id(self, xml_msr_file):
+        """
+        method to link the database id tags between msr.xml file and *.adj file to assign source and epoch to adjusted
+        measurements
+        :param xml_msr_file: input measurement xml file
+        :return:
+        """
+
+        metadata = {}
+
+        # find source/epoch info in msr file
+        with open(xml_msr_file, 'r') as f:
+            for line in f:
+                if '<DnaMeasurement>' in line:
+                    epoch = None
+                    source = None
+                    msr_id = None
+
+                if '<Source>' in line:
+                    source = re.findall("<Source>(.*?)</Source>", line)[0]
+                if '<Epoch>' in line:
+                    epoch = re.findall("<Epoch>(.*?)</Epoch>", line)[0]
+                if '<MeasurementID>' in line:
+                    msr_id = re.findall("<MeasurementID>(.*?)</MeasurementID>", line)[0]
+
+                    if msr_id in metadata:
+                        raise ValueError(f' *** repeat occurance of msr_id: {msr_id}')
+                    if msr_id != '':
+                        metadata[msr_id] = {
+                            'epoch': epoch,
+                            'source': source,
+                        }
+
+        # match with *.adj results using msr id
+        for m in self.msrs:
+            if m.msr_type != 'D':
+                if m.msr_id in metadata:
+                    m.epoch = metadata[m.msr_id]['epoch']
+                    m.source = metadata[m.msr_id]['source']
+            else:
+                if m.msr_id:
+                    if m.msr_id[0] in metadata:
+                        m.epoch = metadata[m.msr_id[0]]['epoch']
+                        m.source = metadata[m.msr_id[0]]['source']
+
+    def write_shp(self, network_name):
```

Comment:
Are you able to add an output directory argument to this method? When I tested it wrote the shapefiles directly into a folder inside the PynAdjust directory

Reply:
Thanks, @BatchelorJ. 

By default, the 'shp' directory is created within the current working directory, where ever that may be. 

As requested, I've added the option to specify where the shapefiles are written.

Enjoy!